// @atlaskit/adf-schema/dist/cjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var schema_1 = require("./schema");
exports.AnnotationTypes = schema_1.AnnotationTypes;
exports.alignment = schema_1.alignment;
exports.alignmentPositionMap = schema_1.alignmentPositionMap;
exports.annotation = schema_1.annotation;
exports.bitbucketSchema = schema_1.bitbucketSchema;
exports.blockCard = schema_1.blockCard;
exports.blockquote = schema_1.blockquote;
exports.bodiedExtension = schema_1.bodiedExtension;
exports.bodiedExtensionWithLocalId = schema_1.bodiedExtensionWithLocalId;
exports.breakout = schema_1.breakout;
exports.bulletList = schema_1.bulletList;
exports.bulletListSelector = schema_1.bulletListSelector;
exports.code = schema_1.code;
exports.codeBlock = schema_1.codeBlock;
exports.codeBlockToJSON = schema_1.codeBlockToJSON;
exports.colorPalette = schema_1.colorPalette;
exports.colorPaletteExperimental = schema_1.colorPaletteExperimental;
exports.confluenceInlineComment = schema_1.confluenceInlineComment;
exports.confluenceJiraIssue = schema_1.confluenceJiraIssue;
exports.confluenceSchema = schema_1.confluenceSchema;
exports.confluenceSchemaWithMediaSingle = schema_1.confluenceSchemaWithMediaSingle;
exports.confluenceUnsupportedBlock = schema_1.confluenceUnsupportedBlock;
exports.confluenceUnsupportedInline = schema_1.confluenceUnsupportedInline;
exports.copyPrivateMediaAttributes = schema_1.copyPrivateMediaAttributes;
exports.createJIRASchema = schema_1.createJIRASchema;
exports.createSchema = schema_1.createSchema;
exports.date = schema_1.date;
exports.decisionItem = schema_1.decisionItem;
exports.decisionList = schema_1.decisionList;
exports.decisionListSelector = schema_1.decisionListSelector;
exports.defaultSchema = schema_1.defaultSchema;
exports.defaultSchemaConfig = schema_1.defaultSchemaConfig;
exports.doc = schema_1.doc;
exports.em = schema_1.em;
exports.embedCard = schema_1.embedCard;
exports.emoji = schema_1.emoji;
exports.expand = schema_1.expand;
exports.expandToJSON = schema_1.expandToJSON;
exports.extension = schema_1.extension;
exports.extensionWithLocalId = schema_1.extensionWithLocalId;
exports.getSchemaBasedOnStage = schema_1.getSchemaBasedOnStage;
exports.hardBreak = schema_1.hardBreak;
exports.heading = schema_1.heading;
exports.image = schema_1.image;
exports.indentation = schema_1.indentation;
exports.inlineCard = schema_1.inlineCard;
exports.inlineExtension = schema_1.inlineExtension;
exports.inlineExtensionWithLocalId = schema_1.inlineExtensionWithLocalId;
exports.inlineNodes = schema_1.inlineNodes;
exports.isSchemaWithAdvancedTextFormattingMarks = schema_1.isSchemaWithAdvancedTextFormattingMarks;
exports.isSchemaWithBlockQuotes = schema_1.isSchemaWithBlockQuotes;
exports.isSchemaWithCodeBlock = schema_1.isSchemaWithCodeBlock;
exports.isSchemaWithEmojis = schema_1.isSchemaWithEmojis;
exports.isSchemaWithLinks = schema_1.isSchemaWithLinks;
exports.isSchemaWithLists = schema_1.isSchemaWithLists;
exports.isSchemaWithMedia = schema_1.isSchemaWithMedia;
exports.isSchemaWithMentions = schema_1.isSchemaWithMentions;
exports.isSchemaWithSubSupMark = schema_1.isSchemaWithSubSupMark;
exports.isSchemaWithTables = schema_1.isSchemaWithTables;
exports.isSchemaWithTextColor = schema_1.isSchemaWithTextColor;
exports.layoutColumn = schema_1.layoutColumn;
exports.layoutSection = schema_1.layoutSection;
exports.link = schema_1.link;
exports.linkToJSON = schema_1.linkToJSON;
exports.listItem = schema_1.listItem;
exports.media = schema_1.media;
exports.mediaGroup = schema_1.mediaGroup;
exports.mediaSingle = schema_1.mediaSingle;
exports.mediaSingleToJSON = schema_1.mediaSingleToJSON;
exports.mediaToJSON = schema_1.mediaToJSON;
exports.mention = schema_1.mention;
exports.mentionToJSON = schema_1.mentionToJSON;
exports.nestedExpand = schema_1.nestedExpand;
exports.orderedList = schema_1.orderedList;
exports.orderedListSelector = schema_1.orderedListSelector;
exports.panel = schema_1.panel;
exports.paragraph = schema_1.paragraph;
exports.placeholder = schema_1.placeholder;
exports.rule = schema_1.rule;
exports.sanitizeNodes = schema_1.sanitizeNodes;
exports.setCellAttrs = schema_1.setCellAttrs;
exports.status = schema_1.status;
exports.strike = schema_1.strike;
exports.strong = schema_1.strong;
exports.subsup = schema_1.subsup;
exports.table = schema_1.table;
exports.tableBackgroundBorderColor = schema_1.tableBackgroundBorderColor;
exports.tableBackgroundColorNames = schema_1.tableBackgroundColorNames;
exports.tableBackgroundColorPalette = schema_1.tableBackgroundColorPalette;
exports.tableCell = schema_1.tableCell;
exports.tableCellContentDomSelector = schema_1.tableCellContentDomSelector;
exports.tableCellContentWrapperSelector = schema_1.tableCellContentWrapperSelector;
exports.tableCellSelector = schema_1.tableCellSelector;
exports.tableHeader = schema_1.tableHeader;
exports.tableHeaderSelector = schema_1.tableHeaderSelector;
exports.tablePrefixSelector = schema_1.tablePrefixSelector;
exports.tableRow = schema_1.tableRow;
exports.tableToJSON = schema_1.tableToJSON;
exports.taskItem = schema_1.taskItem;
exports.taskList = schema_1.taskList;
exports.taskListSelector = schema_1.taskListSelector;
exports.text = schema_1.text;
exports.textColor = schema_1.textColor;
exports.toJSONTableCell = schema_1.toJSONTableCell;
exports.toJSONTableHeader = schema_1.toJSONTableHeader;
exports.typeAheadQuery = schema_1.typeAheadQuery;
exports.underline = schema_1.underline;
exports.unknownBlock = schema_1.unknownBlock;
exports.unsupportedBlock = schema_1.unsupportedBlock;
exports.unsupportedInline = schema_1.unsupportedInline;
exports.unsupportedNodeTypesForMediaCards = schema_1.unsupportedNodeTypesForMediaCards;
exports.buildAnnotationMarkDataAttributes = schema_1.buildAnnotationMarkDataAttributes;
exports.AnnotationMarkStates = schema_1.AnnotationMarkStates;
var utils_1 = require("./utils");
exports.B100 = utils_1.B100;
exports.B400 = utils_1.B400;
exports.B50 = utils_1.B50;
exports.B500 = utils_1.B500;
exports.B75 = utils_1.B75;
exports.DEFAULT_LANGUAGES = utils_1.DEFAULT_LANGUAGES;
exports.G200 = utils_1.G200;
exports.G300 = utils_1.G300;
exports.G400 = utils_1.G400;
exports.G50 = utils_1.G50;
exports.G500 = utils_1.G500;
exports.G75 = utils_1.G75;
exports.N0 = utils_1.N0;
exports.N20 = utils_1.N20;
exports.N200 = utils_1.N200;
exports.N30 = utils_1.N30;
exports.N300 = utils_1.N300;
exports.N40 = utils_1.N40;
exports.N50 = utils_1.N50;
exports.N500 = utils_1.N500;
exports.N60 = utils_1.N60;
exports.N80 = utils_1.N80;
exports.N800 = utils_1.N800;
exports.N90 = utils_1.N90;
exports.P100 = utils_1.P100;
exports.P300 = utils_1.P300;
exports.P400 = utils_1.P400;
exports.P50 = utils_1.P50;
exports.P500 = utils_1.P500;
exports.P75 = utils_1.P75;
exports.R100 = utils_1.R100;
exports.R300 = utils_1.R300;
exports.R400 = utils_1.R400;
exports.R50 = utils_1.R50;
exports.R500 = utils_1.R500;
exports.R75 = utils_1.R75;
exports.T100 = utils_1.T100;
exports.T300 = utils_1.T300;
exports.T50 = utils_1.T50;
exports.T500 = utils_1.T500;
exports.T75 = utils_1.T75;
exports.Y200 = utils_1.Y200;
exports.Y400 = utils_1.Y400;
exports.Y50 = utils_1.Y50;
exports.Y500 = utils_1.Y500;
exports.Y75 = utils_1.Y75;
exports.acNameToEmoji = utils_1.acNameToEmoji;
exports.acShortcutToEmoji = utils_1.acShortcutToEmoji;
exports.createLanguageList = utils_1.createLanguageList;
exports.emojiIdToAcName = utils_1.emojiIdToAcName;
exports.filterSupportedLanguages = utils_1.filterSupportedLanguages;
exports.findMatchedLanguage = utils_1.findMatchedLanguage;
exports.generateUuid = utils_1.generateUuid;
exports.getEmojiAcName = utils_1.getEmojiAcName;
exports.getLanguageIdentifier = utils_1.getLanguageIdentifier;
exports.getLinkMatch = utils_1.getLinkMatch;
exports.hexToRgb = utils_1.hexToRgb;
exports.hexToRgba = utils_1.hexToRgba;
exports.isHex = utils_1.isHex;
exports.isRgb = utils_1.isRgb;
exports.isSafeUrl = utils_1.isSafeUrl;
exports.normalizeHexColor = utils_1.normalizeHexColor;
exports.normalizeUrl = utils_1.normalizeUrl;
exports.rgbToHex = utils_1.rgbToHex;
exports.uuid = utils_1.uuid;
