// @forge/ui/dist/esm/markdown.js
import markdownit from 'markdown-it';
import { MarkdownParser } from 'prosemirror-markdown';
import { defaultSchema } from '@atlaskit/adf-schema';
import { JSONTransformer } from '@atlaskit/editor-json-transformer';
const md = markdownit('zero', { linkify: true, breaks: true }).enable([
    'emphasis',
    'strikethrough',
    'newline',
    'link',
    'linkify',
    'backticks',
]);
const auxMarkdownParser = new MarkdownParser(defaultSchema, md, {
    paragraph: { block: 'paragraph' },
    softbreak: { node: 'hardBreak' },
    hardbreak: { node: 'hardBreak' },
    strong: { mark: 'strong' },
    em: { mark: 'em' },
    s: { mark: 'strike' },
    // eslint-disable-next-line @typescript-eslint/camelcase
    code_inline: { mark: 'code' },
    link: {
        mark: 'link',
        getAttrs: (tok) => ({
            href: tok.attrGet('href'),
            title: tok.attrGet('title') || null,
        }),
    },
});
const jsonTransformer = new JSONTransformer();
export const md2Adf = (md) => {
    const doc = auxMarkdownParser.parse(md);
    return jsonTransformer.encode(doc);
};
