// @forge/ui/dist/cjs/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const tslib_1 = require("tslib");
// pragma
function createElement(type, props, ...children) {
    return {
        type,
        key: null,
        props: {
            ...props,
            children: children.reduce((flattened, child) => {
                if (child) {
                    return flattened.concat(child);
                }
                return flattened;
            }, []),
        },
    };
}
var backend_runtime_1 = require("./backend-runtime");
exports.render = backend_runtime_1.render;
var useAction_1 = require("./hooks/useAction");
exports.useAction = useAction_1.useAction;
var useState_1 = require("./hooks/useState");
exports.useState = useState_1.useState;
var useProductContext_1 = require("./hooks/useProductContext");
exports.useProductContext = useProductContext_1.useProductContext;
var useConfig_1 = require("./hooks/useConfig");
exports.useConfig = useConfig_1.useConfig;
tslib_1.__exportStar(require("./components"), exports);
exports.default = { createElement };
