// @atlaskit/adf-schema/dist/cjs/steps/table/utils/side-effects/table.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var prosemirror_transform_1 = require("prosemirror-transform");
var TableSideEffectHandler = /** @class */ (function () {
    function TableSideEffectHandler(tableSideEffect) {
        if (tableSideEffect) {
            this.table = tableSideEffect;
        }
    }
    TableSideEffectHandler.prototype.addTableSideEffect = function (from, to, node) {
        this.table = { from: from, to: to, node: node };
    };
    TableSideEffectHandler.prototype.handleAddTable = function (tr, isDelete) {
        if (isDelete || !this.table) {
            return false;
        }
        tr.insert(this.table.from, this.table.node);
        return true;
    };
    TableSideEffectHandler.prototype.handleRemoveTable = function (tr, tablePos, tableRect, column, isDelete) {
        if (isDelete && tableRect.map.width === 1 && column === 0) {
            // Add side effect
            this.addTableSideEffect(tablePos, tablePos + tableRect.table.nodeSize, tableRect.table.copy(tableRect.table.content));
            tr.delete(tablePos, tablePos + tableRect.table.nodeSize);
            return true;
        }
        return false;
    };
    TableSideEffectHandler.prototype.getTableMap = function (isDelete) {
        if (!this.table) {
            return;
        }
        var _a = this.table, from = _a.from, to = _a.to;
        if (isDelete) {
            return new prosemirror_transform_1.StepMap([from, to - from, 0]);
        }
        return new prosemirror_transform_1.StepMap([from, 0, to - from]);
    };
    TableSideEffectHandler.prototype.map = function (mapping) {
        if (!this.table) {
            return;
        }
        return {
            from: mapping.map(this.table.from),
            to: mapping.map(this.table.to),
            node: this.table.node,
        };
    };
    TableSideEffectHandler.prototype.invert = function (doc) {
        if (!this.table) {
            return;
        }
        var tableNode = doc.nodeAt(this.table.from);
        return {
            from: this.table.from,
            to: this.table.from + tableNode.nodeSize,
            node: tableNode.copy(tableNode.content),
        };
    };
    TableSideEffectHandler.prototype.toJSON = function () {
        if (!this.table) {
            return;
        }
        return {
            from: this.table.from,
            to: this.table.to,
            node: this.table.node.toJSON(),
        };
    };
    TableSideEffectHandler.fromJSON = function (schema, json) {
        return {
            from: json.from,
            to: json.to,
            node: schema.nodeFromJSON(json.node),
        };
    };
    return TableSideEffectHandler;
}());
exports.TableSideEffectHandler = TableSideEffectHandler;
