// @atlaskit/adf-schema/dist/cjs/schema/marks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var em_1 = require("./em");
exports.em = em_1.em;
var code_1 = require("./code");
exports.code = code_1.code;
var strike_1 = require("./strike");
exports.strike = strike_1.strike;
var strong_1 = require("./strong");
exports.strong = strong_1.strong;
var underline_1 = require("./underline");
exports.underline = underline_1.underline;
var link_1 = require("./link");
exports.link = link_1.link;
exports.linkToJSON = link_1.toJSON;
var type_ahead_query_1 = require("./type-ahead-query");
exports.typeAheadQuery = type_ahead_query_1.typeAheadQuery;
var subsup_1 = require("./subsup");
exports.subsup = subsup_1.subsup;
var text_color_1 = require("./text-color");
exports.textColor = text_color_1.textColor;
exports.colorPalette = text_color_1.colorPalette;
exports.colorPaletteExperimental = text_color_1.colorPaletteExperimental;
var confluence_inline_comment_1 = require("./confluence-inline-comment");
exports.confluenceInlineComment = confluence_inline_comment_1.confluenceInlineComment;
var breakout_1 = require("./breakout");
exports.breakout = breakout_1.breakout;
var alignment_1 = require("./alignment");
exports.alignment = alignment_1.alignment;
exports.alignmentPositionMap = alignment_1.alignmentPositionMap;
var indentation_1 = require("./indentation");
exports.indentation = indentation_1.indentation;
var annotation_1 = require("./annotation");
exports.annotation = annotation_1.annotation;
exports.AnnotationMarkStates = annotation_1.AnnotationMarkStates;
exports.buildAnnotationMarkDataAttributes = annotation_1.buildDataAtributes;
exports.AnnotationTypes = annotation_1.AnnotationTypes;
