// @atlaskit/adf-schema/dist/cjs/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var emoji_1 = require("./confluence/emoji");
exports.acNameToEmoji = emoji_1.acNameToEmoji;
exports.acShortcutToEmoji = emoji_1.acShortcutToEmoji;
exports.emojiIdToAcName = emoji_1.emojiIdToAcName;
exports.getEmojiAcName = emoji_1.getEmojiAcName;
var uuid_1 = require("./uuid");
exports.generateUuid = uuid_1.generateUuid;
exports.uuid = uuid_1.uuid;
var colors_1 = require("./colors");
exports.B100 = colors_1.B100;
exports.B400 = colors_1.B400;
exports.B50 = colors_1.B50;
exports.B500 = colors_1.B500;
exports.B75 = colors_1.B75;
exports.G200 = colors_1.G200;
exports.G300 = colors_1.G300;
exports.G400 = colors_1.G400;
exports.G50 = colors_1.G50;
exports.G500 = colors_1.G500;
exports.G75 = colors_1.G75;
exports.N0 = colors_1.N0;
exports.N20 = colors_1.N20;
exports.N200 = colors_1.N200;
exports.N30 = colors_1.N30;
exports.N300 = colors_1.N300;
exports.N40 = colors_1.N40;
exports.N50 = colors_1.N50;
exports.N500 = colors_1.N500;
exports.N60 = colors_1.N60;
exports.N80 = colors_1.N80;
exports.N800 = colors_1.N800;
exports.N90 = colors_1.N90;
exports.P100 = colors_1.P100;
exports.P300 = colors_1.P300;
exports.P400 = colors_1.P400;
exports.P50 = colors_1.P50;
exports.P500 = colors_1.P500;
exports.P75 = colors_1.P75;
exports.R100 = colors_1.R100;
exports.R300 = colors_1.R300;
exports.R400 = colors_1.R400;
exports.R50 = colors_1.R50;
exports.R500 = colors_1.R500;
exports.R75 = colors_1.R75;
exports.T100 = colors_1.T100;
exports.T300 = colors_1.T300;
exports.T50 = colors_1.T50;
exports.T500 = colors_1.T500;
exports.T75 = colors_1.T75;
exports.Y200 = colors_1.Y200;
exports.Y400 = colors_1.Y400;
exports.Y50 = colors_1.Y50;
exports.Y500 = colors_1.Y500;
exports.Y75 = colors_1.Y75;
exports.hexToRgb = colors_1.hexToRgb;
exports.hexToRgba = colors_1.hexToRgba;
exports.isHex = colors_1.isHex;
exports.isRgb = colors_1.isRgb;
exports.normalizeHexColor = colors_1.normalizeHexColor;
exports.rgbToHex = colors_1.rgbToHex;
var url_1 = require("./url");
exports.getLinkMatch = url_1.getLinkMatch;
exports.isSafeUrl = url_1.isSafeUrl;
exports.normalizeUrl = url_1.normalizeUrl;
var languageList_1 = require("./languageList");
exports.DEFAULT_LANGUAGES = languageList_1.DEFAULT_LANGUAGES;
exports.createLanguageList = languageList_1.createLanguageList;
exports.filterSupportedLanguages = languageList_1.filterSupportedLanguages;
exports.findMatchedLanguage = languageList_1.findMatchedLanguage;
exports.getLanguageIdentifier = languageList_1.getLanguageIdentifier;
